lib/bowling_game.rb

```diff
@@ -0,0 +1,58 @@
+class BowlingGame
+
+  MAX_ROLLS = 20
+
+  attr_reader :rolls, :result
+
+  def initialize
+    @rolls = []
+    @result = 0
+    @frame = 1
+  end
+
+  def roll(pins)
+    @rolls << pins
+  end
+
+  def score
+    roll_index = 0
+    raise "The game has finished, final score is #{@result}." if @frame > 10
+    10.times do
+      if strike?(roll_index)
+        @result += @rolls[roll_index] + @rolls[roll_index + 1] + @rolls[roll_index + 2]
+        roll_index += 1
+      elsif spare?(roll_index)
+        @result += spare_score(roll_index)
+        roll_index += 2
+      else
+        @result += frame_score(roll_index)
+        roll_index += 2
+      end
+      @frame += 1
+    end
+    result
+  end
+
+  private
```

Comment:
Really like the idea of putting these methods into private. 

Also means fewer tests.